lib/erblint-github/linters/github/accessibility/avoid_generic_link_text_counter.rb

```diff
@@ -36,29 +37,59 @@ def run(processed_source)
                 prev_node_tag = BetterHtml::Tree::Tag.from_node(prev_node)
                 next_node_tag = BetterHtml::Tree::Tag.from_node(next_node)
 
-                # We only report if the text is nested between two link tags.
+                aria_label = possible_attribute_values(prev_node_tag, "aria-label")
+                aria_labelledby = possible_attribute_values(prev_node_tag, "aria-labelledby")
+
+                # Checks if nested between two link tags.
                 if link_tag?(prev_node_tag) && link_tag?(next_node_tag) && next_node_tag.closing?
+                  # Skip because we cannot reliably check accessible name from aria-labelledby, or an aria-label that is set to a variable
+                  # with static code analysis.
+                  next if aria_labelledby.present? || (aria_label.present? && aria_label.join.include?("<%="))
```

Comment:
We skip flagging the link if it has `aria-labelledby` or an aria-label that is set to a variable like:

```.erb
<a href="github.com" aria-label="<%= some_variable%>" >
```
since it may be acceptable usecase and we cannot evaluate with linter.